Site/ASAB Maestro/Descriptors/elasticsearch.yaml

```diff
@@ -38,3 +38,59 @@ descriptor:
     xpack.security.transport.ssl.certificate_authorities: certs/ca.pem
     xpack.security.transport.ssl.certificate: certs/certificate.pem
     xpack.security.transport.ssl.key: certs/key.pem
+
+
+nginx:
+  proxy_caches:
+    elasticsearch: keys_zone=elasticsearch:1m max_size=2m
+
+  https:
+    location /api/elasticsearch/:
+      - auth_request /_elasticsearch_introspect
+      - auth_request_set $batman $upstream_http_authorization
+      - proxy_set_header Authorization $batman
+
+      - auth_request_set $cookie $upstream_http_cookie
+      - proxy_set_header Cookie $cookie
+
+      - rewrite ^/api/elasticsearch/(.*) /$1 break
+      - proxy_set_header Host $host
+      - proxy_set_header Connection "Keep-Alive"
+      - proxy_set_header Proxy-Connection "Keep-Alive"
+      - proxy_pass http://elasticsearch  # upstream created by remote control technology - it leads to master
+
+    location /_elasticsearch_introspect:
+    - internal
+    - proxy_method POST
+    - proxy_set_body ""
+    - proxy_set_header X-Request-URI "request_uri"
+    - proxy_cache elasticsearch
+    - proxy_pass http://upstream-seacat-auth-private/nginx/introspect/batman?client_id=elasticsearch
+    - proxy_cache_key $http_authorization
+    - proxy_cache_lock on
+    - proxy_cache_valid 200 10s
+    - proxy_ignore_headers Cache-Control Expires Set-Cookie
+
+seacat-auth:
+  content:
+  - "cl.json": |
+      [{
+        "_id": "elasticsearch",
+        "application_type": "api",
+        "authorize_anonymous_users": false,
+        "client_name": "Elasticsearch",
+        "code_challenge_method": "none",
+        "grant_types": [
+          "authorization_code"
+        ],
+        "redirect_uri_validation_method": "prefix_match",
+        "redirect_uris": [
+          "{{PUBLIC_URL}}/api/elasticsearch"
```

Comment:
Are these needed?

... it prevents use of the file instead of in-place `cl.json`.


Not a big deal, anyway.